### ISSUE TITLE
documentationUrl not used in reports

### ISSUE THREAD

**Issue by liveaverage** [daveshanley/vacuum#416]
Despite setting `documentationUrl` in the ruleset, the generated html report still references quobix.com:

![image](https://github.com/daveshanley/vacuum/assets/3045804/bbad5d32-917f-43b2-8f01-31998b5b9834)

Sample ruleset:

```
extends: [[spectral:oas, off]]
documentationUrl: "https://my-url.com/custom-rulesets#documentation-url"
rules:
  operation-operationId: true
  info-contact: true
```

Sample spec:

```{
  "openapi": "3.0.1",
  "info": {
    "title": "Test Service",
    "termsOfService": "https://www.mysite.com/en-us/legal_info",
    "version": "v2.61.11"
  },
  "servers": [
    {
      "url": "https://stg.mysite.com",
      "description": "Generated server url"
    }
  ],
...
```

Is there another parameter that should be set? Is this the only area using `documentationUrl` for linting errors, or should we see this referenced in CLI output as well?

**Comment by daveshanley** [comment]
This is the first use-case I have heard where someone would like to change the URL in the report. Happy to add this in, could you tell me more about how you would like to use the URL and what you would expect in the html-report and other commands?

**Comment by liveaverage** [comment]
Absolutely! This is a gap I noticed in Spectral as well... they actually reference a common use case for `documentationUrl` but I don't see it actually used anywhere [within their core product] as described. I'm guessing it's consumed as a part of their Studio product, but we typically provide governance & reporting via pipelines: https://docs.stoplight.io/docs/spectral/e5b9616d6d50c-rulesets#documentation-url

A couple ideas come to mind with regard to html-report:
- Template `documentationUrl` + an anchor link corresponding to the rule name, which would allow a quick reference to a more robust & informative API style guide entry
- As an alternative (or addition), the ability to reference `documentationUrl` as a template parameter within the rule `description`, `message,` or custom `howToFix` field would be amazing and allow references in or out of html-reports, independent of the "Learn more about:" section.

**Comment by daveshanley** [comment]
Interesting, sounds like a nice upgrade.
